cloud/src/recycler/hdfs_accessor.cpp

```diff
@@ -357,7 +400,15 @@ int HdfsAccessor::delete_prefix(const std::string& path_prefix, int64_t expirati
     auto uri = to_uri(path_prefix);
     LOG(INFO) << "delete prefix, uri=" << uri;
     std::unique_ptr<ListIterator> list_iter;
-    int ret = list_all(&list_iter);
+    auto tablet_path = extract_tablet_path(path_prefix);
```

Comment:
add comment

Reply:
list prefix first, if not match, backoff to list the parent folder to match the prefix